source/MaterialXCore/Interface.cpp

```diff
@@ -116,37 +128,29 @@ OutputPtr PortElement::getConnectedOutput() const
     else if (hasNodeName())
     {
         const string& nodeName = getNodeName();
-        ConstElementPtr startingElement = getParent();
-        if (startingElement)
+        NodePtr node = resolveRootNameReference<Node>(nodeName, root);
```

Comment:
This now just mirrors the nodegraph code as the only difference is that it's looking for a node vs nodegraph. Previous code would miss siblings outside of nodegraphs.